akka-sample-spring-extension-java/src/main/java/org/example/akka/AkkaConfig.java

```diff
@@ -0,0 +1,30 @@
+package org.example.akka;
+
+import akka.actor.ActorSystem;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+import org.example.akka.di.SpringExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class AkkaConfig {
+
+	@Autowired
+	private ApplicationContext applicationContext;
+	
+	@Bean(destroyMethod = "terminate")
+	public ActorSystem actorSystem() {
+		ActorSystem actorSystem = ActorSystem.create("akka-system", akkaConfiguration());
+		SpringExtension.SpringExtProvider.get(actorSystem).initialize(applicationContext);
```

Comment:
This means that there is a potential short time when the SpringExtProvider is not yet initialised and trying to create an actor using it would fail with a NPE.

This could be made completely safe byt passing the applicationContext to the extension as a `akka.actor.setup.Setup` instead, and pick it up from the actor system in `createExtension`. That way the application context field of the extension could also be final instead of having to be volatile.

Reply:
Hm, at this point I am a little lost. I'm very new to the way Akka works. I understand that the ActorSystem is initialized before the Extension, thus, the ActorSystem could try to create an Actor  using the Extension just to find that the ApplicationContext is still null.

If so, I'm lost here because the initialization of the ActorSystem would depend on the initialization of the Extension's ApplicationContext, and viceversa.

Could I get a little extra help to further improve the PR?

Reply:
The `Setup` API allows you to define and pass objects into the actorsystem that is then available through `ActorSystem.setup`. Create your own Setup subclass which takes the application context to its constructor and keeps it as a final field. Something like this:

```
class SpringSetup extends Setup { ... }

// where you create the actor system
ActorSystem system = ActorSystem.create("system-name", 
        ActorSystemSetup.create(new SpringSetup(applicationContext)));
    
// in the extension
ApplicationContext ctx = system.settings().setup().get(SpringSetup.class)
  .orElseThrow(new RuntimeException("ActorSystem must be created with a SpringSetup"))
  .getApplicationContext
```